lib/improver/orographic_enhancement.py

```diff
@@ -0,0 +1,600 @@
+
+        Args:
+            orogenh_data (np.ndarray):
+                Orographic enhancement value in mm h-1
+            reference_cube (iris.cube.Cube):
+                Cube with the correct time and forecast period coordinates on
+                the UK 2 km standard grid
+
+        Returns:
+            (tuple): tuple containing:
+                **orogenh** (iris.cube.Cube):
+                    Orographic enhancement cube on 1 km UKPP grid
+                **orogenh_standard_grid** (iris.cube.Cube):
+                    Orographic enhancement cube on 2 km standard grid, padded
+                    with masked np.nans
+        """
+        # create cube containing high resolution data in mm/h
+        x_coord = self.topography.coord(axis='x')
+        y_coord = self.topography.coord(axis='y')
+        attributes = {'institution': 'Met Office', 'source': 'IMPROVER'}
+        orogenh = iris.cube.Cube(
+            orogenh_data, long_name="orographic_enhancement",
+            units="mm h-1", attributes=attributes,
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+        orogenh.add_aux_coord(reference_cube.coord('time'))
+        orogenh.add_aux_coord(reference_cube.coord('forecast_reference_time'))
+        orogenh.add_aux_coord(reference_cube.coord('forecast_period'))
+
+        # regrid the orographic enhancement cube onto the standard grid and
+        # mask extrapolated points
+        orogenh_standard_grid = orogenh.regrid(
+            reference_cube, iris.analysis.Linear(extrapolation_mode='mask'))
+
+        for axis in ['x', 'y']:
+            orogenh_standard_grid = sort_coord_in_cube(
+                orogenh_standard_grid, orogenh_standard_grid.coord(axis=axis))
+
+        return orogenh, orogenh_standard_grid
+
+    def process(self, temperature, humidity, pressure, uwind, vwind,
+                topography, svp_method='IMPROVER'):
+        """
+        Calculate precipitation enhancement over orography on standard (2 km)
+        and high resolution (1 km UKPP domain) grids
+
+        Args:
+            temperature (iris.cube.Cube):
+                Temperature at top of boundary layer
+            humidity (iris.cube.Cube):
+                Relative humidity at top of boundary layer
+            pressure (iris.cube.Cube):
+                Pressure at top of boundary layer
+            uwind (iris.cube.Cube):
+                Positive eastward wind vector component at top of boundary
+                layer
+            vwind (iris.cube.Cube):
+                Positive northward wind vector component at top of boundary
+                layer
+            topography (iris.cube.Cube):
+                Height of topography above sea level on 1 km UKPP domain grid
+
+        Kwargs:
+            svp_method (str):
+                Which method to use to calculate the saturation vapour
+                pressure (IMPROVER or STEPS).  Defaults to using the IMPROVER
+                WetBulbTemperature plugin.
```

Comment:
Is there any reason to keep the STEPS method? It would make the code a little cleaner to discard it now that you've proved it makes no difference.

Reply:
As discussed - will get rid of this once the demonstration has been second-reviewed.